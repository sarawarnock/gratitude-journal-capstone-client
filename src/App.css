html {
  /* background-color: #F2E3CE; */
  background: url(background.jpeg) no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

.App {
  text-align: center;
  font-family: 'Open Sans Condensed', sans-serif;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.button {
  display: inline-flex;
  height: 40px;
  width: 150px;
  border: 2px solid #BFC0C0;
  margin: 20px 20px 20px 20px;
  color: #BFC0C0;
  /* text-transform: uppercase; */
  text-decoration: none;
  font-size: 1em;
  letter-spacing: 1.5px;
  align-items: center;
  justify-content: center;
  overflow: hidden;
}

a {
  color: #BFC0C0;
  text-decoration: none;
  letter-spacing: 1px;
}

#button-3 {
  position: relative;
  overflow: hidden;
  cursor: pointer;
}

#button-3 a {
  position: relative;
  transition: all .45s ease-Out;
}

#circle {
  width: 0%;
  height: 0%;
  opacity: 0;
  line-height: 40px;
  border-radius: 50%;
  background: #BFC0C0;
  position: absolute;
  transition: all .5s ease-Out;
  top: 20px;
  left: 70px;
}

#button-3:hover #circle {
  width: 200%;
  height: 500%;
  opacity: 1;
  top: -70px;
  left: -70px;
}

#button-3:hover a {
  color: #2D3142;
}

h1 {
  font-family: 'Shadows Into Light Two', cursive;
  font-size: 44px;
  color: black;
  padding: 0px;
  margin: 5px;
}

h2 {
  /* color: #EBA29D; */
  color: black;
  /* color: #023B09; */
  font-size: 28px;
  margin: 0px;
}

p {
  font-size: 20px;
  color: black;
}

a {
  color: black;
  text-decoration: none;
}

a:active {
  text-decoration: underline;
  /* text-decoration-color: #EBA29D; */
  text-decoration-color: #F2E3CE;
}

a:hover {
  text-decoration: underline;
  text-decoration-color: #F2E3CE;
}

/* .navbar {
  background-color: #F2E3CE;
} */

.menu {
  display: flex;
  flex-direction: column;
  align-content: center;
}

.menu, li {
  align-content: center;
}

.nav-token {
  padding: 10px;
  margin: 0;
}

.nav-link {
  list-style-type: none;
  font-size: 22px;
}

.new {
  background-color: #F2E3CE;
  margin: 5px;
}

.new-entry {
  background-color: #F2E3CE;
  opacity: .8;
}

.home {
  background-color: #F2E3CE;
  margin: 5px;
}

ul {
  font-size: 16px;
  padding: 0px;
}

li {
  font-size: 18px;
  list-style-type: none;
}

button {
  cursor: pointer;
}

/* button:hover {
  /* background-color: #EBA29D
  background-color: #F2E3CE;
} */

/* .submit-button {
  color: black;
} */

.landing__container {
  display: flex;
  flex-direction: column;
  text-align: center;
  border: 1px solid white;
  padding: 10px;
  margin: 10px;
}

.landing__container, p {
  align-items: center;
}

.login__container {
  display: flex;
  text-align: center;
  /* border: 1px solid white; */
  margin: 10px;
  padding: 10px;
  background-color: #F2E3CE;
  opacity: .8;
}

.login__container, p {
  flex-direction: column;
  align-items: center;
  margin: 3px;
  padding: 0px;
}

input {
  margin: 5px;
}

.overall-mood {
  margin: 10px;
  padding: 10px;
  color: black;
  font-size: 20px;
}

.entry-name {
  color: black;
  font-size: 20px;
}

/* entry__container {
  background-color: #F2E3CE;
  opacity: .7;
} */

.entry {
  color: black;
  background-color: #F2E3CE;
  opacity: .7;
}

@media screen and (min-width: 600px) {
  .App {
    display: grid;
    grid-template-columns: 15% 35% 35% 15%;
    /* grid-template-rows: 1fr; */
    grid-template-areas: 
    "nav nav header header "
    ". main main ."
    "journal journal journal journal"
    ;
  }

  header {
    grid-area: header;
  }

  .menu {
    grid-area: nav;
  }

  main {
    grid-area: main;
  }

  .main {
    grid-area: main;
  }

  .my-journal {
    grid-area: journal;
  }

  .my-journal-entries {
    display: block;
    grid-area: journal;
    grid-row-gap: 10px;
  }
}
