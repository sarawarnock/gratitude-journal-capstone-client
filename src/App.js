import React from 'react';
import { Route, Link, Switch } from 'react-router-dom'
import './App.css'
import config from './config'
import TokenService from './services/token-service'
import Navbar from './components/navbar'
import Header from './components/header'
import Login from './components/login'
import SignUp from './components/signup'
import LandingPage from './components/landing-page'
import NewEntry from './components/new-entry'
import AllEntries from './components/all-entries'
import NotFoundPage from './components/not-found'


export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user_name: '',
      password: '',
      first_name: '',
      last_name: '',
      allEntries: [],
      error: ''
    }
  }

  //component did mount to fetch all entries to pass down to other components
  componentDidMount() {
    let getEntriesUrl = ''
    if (sessionStorage.user_id == undefined) {
      getEntriesUrl = `${config.API_ENDPOINT}/entries`
    } else {
      getEntriesUrl = `${config.API_ENDPOINT}/entries/user/${TokenService.getUserId()}`;
    }

    console.log(sessionStorage.user_id)
    fetch(getEntriesUrl)
      .then(response => response.json())
      .then(entries => {
        this.setState({
          allEntries: entries
        });
        console.log(this.state.allEntries)
      })
      .catch(error => this.setState({ error }))
  }

  //renders the Navbar
  renderNav() {
    return (
      <div>
        <Navbar />
      </div>
    )
  }

  //renders the main pages
  renderMainPages = () => {
    console.log(this.state)
    return (
      <div className="main-pages">
        <Switch>
          <Route 
            exact
            path='/'
            component={LandingPage}
          />
          <Route 
            exact
            path='/signup'
            component={SignUp}
          />
          <Route 
            exact
            path='/login'
            component={Login}
          />
          <Route 
            exact
            path='/home'
            render={(props) => <AllEntries {...props} allEntries={this.state.allEntries} />}
          />
          <Route 
            exact
            path='/newentry'
            component={NewEntry}
          />
          <Route 
            component={NotFoundPage}
          />
        </Switch>
      </div>
    )
  }

  render() {
    return (
      <div className="App">
        <nav className="app-nav"> {this.renderNav()} </nav>
        <header>
          <Link to='/'>
            <Header />
          </Link>
        </header>

        <main>
          {this.renderMainPages()}
        </main>
      </div>
    );
  }
}
