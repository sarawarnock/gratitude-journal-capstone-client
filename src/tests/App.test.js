import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom'
import App from '../app'
import NewEntry from '../components/new-entry'
import AllEntries from '../components/all-entries'
import Login from '../components/login'
import LandingPage from '../components/landing-page'
import SignUp from '../components/signup'
import Header from '../components/header'
import Navbar from '../components/navbar'
import NotFound from '../components/not-found'

const testEntriesArray = [
  {
    id: 1,
    user_id: 1,
    bullet_1: "Test 1",
    bullet_2: "Test 2",
    bullet_3: "Test 3",
    mood: "happy",
    is_public: 0
  },
  {
    id: 2,
    user_id: 1,
    bullet_1: "Test 1",
    bullet_2: "Test 2",
    bullet_3: "Test 3",
    mood: "happy",
    is_public: 0
  }
]

describe('Gratiplicity app', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('New Entry component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <NewEntry />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <NewEntry />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})

describe('All Entries componenet', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <AllEntries allEntries={testEntriesArray} />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AllEntries allEntries={testEntriesArray} />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Login component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Sign Up component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <SignUp />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <SignUp />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Not Found component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <NotFound />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Header component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Navbar component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('Landing Page component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <BrowserRouter>
      <LandingPage />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <LandingPage />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});