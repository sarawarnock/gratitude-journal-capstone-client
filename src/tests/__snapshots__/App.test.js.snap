// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Entries componenet renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <h1>
      My Journal
    </h1>
    <div>
      <h2 />
      <p>
         
        Test 1
         
      </p>
      <p>
         
        Test 2
         
      </p>
      <p>
         
        Test 3
         
      </p>
      <p>
         
        happy
         
      </p>
    </div>
    <div>
      <h2 />
      <p>
         
        Test 1
         
      </p>
      <p>
         
        Test 2
         
      </p>
      <p>
         
        Test 3
         
      </p>
      <p>
         
        happy
         
      </p>
    </div>
    <a
      href="/newentry"
      onClick={[Function]}
    >
      <button>
        New Entry
      </button>
    </a>
  </main>
</div>
`;

exports[`Header component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <h1>
      Gratiplicity: the daily gratitude journal
    </h1>
  </main>
</div>
`;

exports[`Landing Page component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <h1>
      Gratiplicity: Your Daily Gratitude Jounal
    </h1>
    <p>
      Keep track of the things you're grateful for, simply.
    </p>
    <p>
      Jot down 3 things that made you smile each day and keep track of your overall mood.
    </p>
    <button
      className="big-btn"
    >
       
      <a
        href="/signup"
        onClick={[Function]}
      >
         Sign Up 
      </a>
    </button>
    <button
      className="big-btn"
    >
       
      <a
        href="/login"
        onClick={[Function]}
      >
         Login
      </a>
       
    </button>
    <p>
      To login: 
    </p>
    <p>
      username: testuser
    </p>
    <p>
      password: Testpassword1
    </p>
  </main>
</div>
`;

exports[`Login component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <h1>
      Log In
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <label
        className="user-label"
        htmlFor="username"
      >
        Username
      </label>
      <input
        id="username"
        name="loginUsername"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
      />
      <label
        className="user-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="loginPassword"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
      <button
        className="small-btn"
        type="submit"
      >
         Log In 
      </button>
    </form>
    <div>
      <h2>
        Don't have an account yet?
      </h2>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="small-btn"
        >
          Sign Up
        </button>
      </a>
    </div>
  </main>
</div>
`;

exports[`Navbar component renders the UI as expected 1`] = `
<div
  className="navbar"
>
  <nav
    className="nav"
    role="navigation"
  >
    
  </nav>
</div>
`;

exports[`New Entry component renders the UI as expected 1`] = `
<div
  className="App"
>
  <div>
    <form>
      <div
        className="gratitude-values"
      >
        <h2>
           What are you grateful for today? 
        </h2>
        <ul>
          <li>
            <input
              name="gratitudeValue"
              placeholder="#1"
              type="text"
            />
          </li>
          <li>
            <input
              name="gratitudeValue"
              placeholder="#2"
              type="text"
            />
          </li>
          <li>
            <input
              name="gratitudeValue"
              placeholder="#3"
              type="text"
            />
          </li>
        </ul>
      </div>
      <div
        className="overall-mood"
      >
        <h2>
          What is your overall mood?
        </h2>
        <input
          id="happy"
          name="overallMood"
          type="radio"
          value="Happy"
        />
        <label
          htmlFor="happy"
        >
          Happy
        </label>
        <input
          id="ok"
          name="overallMood"
          type="radio"
          value="OK"
        />
        <label
          htmlFor="ok"
        >
          OK
        </label>
        <input
          id="sad"
          name="overallMood"
          type="radio"
          value="Sad"
        />
        <label
          htmlFor="sad"
        >
          Sad
        </label>
      </div>
      <br />
      <div>
        <button
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Not Found component renders the UI as expected 1`] = `
<div>
  <h1>
    Page Not Found!
  </h1>
</div>
`;

exports[`Sign Up component renders the UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <h1>
      Sign Up
    </h1>
    <form
      className="signup-form"
      onSubmit={[Function]}
    >
      <div
        className="signup-email"
      >
        <label
          className="user-label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="signUpEmail"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div
        className="signup-password"
      >
        <label
          className="user-label"
          htmlFor="password"
        >
          Password (one capital letter and one number needed)
        </label>
        <input
          id="password"
          name="signUpPassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="signup-name"
      >
        <label
          className="user-label"
          htmlFor="fname"
        >
          First Name
        </label>
        <input
          id="fname"
          name="signUpFirstName"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          type="text"
        />
      </div>
      <button
        className="small-btn"
        type="submit"
      >
        Sign Up!
      </button>
    </form>
    <div>
      <h2>
        Already have an account?
      </h2>
      <button
        className="small-btn"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </button>
    </div>
  </main>
</div>
`;
